daisy_workflows/image_import/inspection/src/boot_inspect/inspection.py

```diff
@@ -117,6 +117,42 @@ def inspect_device(g, device: str) -> model.InspectionResults:
   )
 
 
+def inspect_boot_loader(g) -> (bool, bool, str):
+  """Finds boot-loader properties for the device using offline inspection.
+
+  Args:
+    g (guestfs.GuestFS): A launched, but unmounted, GuestFS instance.
+
+  Example:
+
+    g = guestfs.GuestFS(python_return_dict=True)
+    g.add_drive_opts("/dev/sdb", format="raw")
+    g.launch()
+    results = inspect_boot_loader(g)
+  """
+
+  bios_bootable = False
+  uefi_bootable = False
+  root_fs = ""
+
+  try:
+    part_list = g.part_list('/dev/sda')
+    for part in part_list:
+      guid = g.part_get_gpt_type('/dev/sda', part['part_num'])
+
+      # It covers both GPT "EFI Sstem" and BIOS "EFI (FAT-12/16/32)"
+      if guid == 'C12A7328-F81F-11D2-BA4B-00A0C93EC93B':
+        uefi_bootable = True
+        # TODO: detect root_fs (b/169245755)
+      if guid == '21686148-6449-6E6F-744E-656564454649':
+        bios_bootable = True
```

Comment:
Recall that there are  four combos of firmware and partition scheme: [UEFI, BIOS]  with [MBR, GPT]. Here you cover only GPT.

My key concern is that anything that was previously bootable without UEFI_COMPATIBLE should not be marked as UEFI_COMPATIBLE. I don't know whether BIOS on GPT is currently bootable; if so, that'd be good to inspect for.

What you really want to identify is whether the inspected OS is bootable by BIOS on MBR.

Good reading:
 - https://wiki.archlinux.org/index.php/partitioning#Choosing_between_GPT_and_MBR
 - https://www.rodsbooks.com/gdisk/hybrid.html
 - https://systemd.io/DISCOVERABLE_PARTITIONS/